lib/puppet/trusted_external.rb

```diff
@@ -3,11 +3,41 @@ module Puppet::TrustedExternal
   def retrieve(certname)
     command = Puppet[:trusted_external_command]
     return nil unless command
+    Puppet.debug _("Retrieving trusted external data from %{command}") % {command: command}
+    setting_type = Puppet.settings.setting(:trusted_external_command).type
+    if setting_type == :file
+      return fetch_data(command, certname)
+    end
+    # command is a directory. Thus, data is a hash of <basename> => <data> for
+    # each executable file in command. For example, if the files 'servicenow.rb',
+    # 'unicorn.sh' are in command, then data is the following hash:
+    #   { 'servicenow' => <servicenow.rb output>, 'unicorn' => <unicorn.sh output> }
+    data = {}
+    Puppet::FileSystem.children(command).each do |file|
+      next unless Puppet::FileSystem.file?(file)
+      abs_path = Puppet::FileSystem.expand_path(file)
```

Comment:
Also thinking we should expand the path first, and then check if the `abs_path` is a `file?`. If not print a debug message before "next"-ing.

Reply:
Updated. Out of curiosity, why use `abs_path`?

Reply:
I was thinking it would be helpful to test (`executable?`, etc) the same path that we display in the error message. It can be confusing if puppet tests a relative path, but displays an absolute path, because maybe the current working directory when the test occurs isn't the same as when the debug message is logged.

Reply:
Makes sense, thanks.